x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/action/filter/ShardBulkInferenceActionFilter.java

```diff
@@ -0,0 +1,534 @@
+                            var acc = inferenceResults.get(request.index);
+                            if (result instanceof ErrorChunkedInferenceResults error) {
+                                acc.addFailure(
+                                    new ElasticsearchException(
+                                        "Exception when running inference id [{}] on field [{}]",
+                                        error.getException(),
+                                        inferenceProvider.model.getInferenceEntityId(),
+                                        request.field
+                                    )
+                                );
+                            } else {
+                                acc.addOrUpdateResponse(
+                                    new FieldInferenceResponse(
+                                        request.field(),
+                                        request.input(),
+                                        request.inputOrder(),
+                                        request.isOriginalFieldInput(),
+                                        inferenceProvider.model,
+                                        result
+                                    )
+                                );
+                            }
+                        }
+                    } finally {
+                        onFinish();
+                    }
+                }
+
+                @Override
+                public void onFailure(Exception exc) {
+                    try {
+                        for (int i = 0; i < requests.size(); i++) {
+                            var request = requests.get(i);
+                            addInferenceResponseFailure(
+                                request.index,
+                                new ElasticsearchException(
+                                    "Exception when running inference id [{}] on field [{}]",
+                                    exc,
+                                    inferenceProvider.model.getInferenceEntityId(),
+                                    request.field
+                                )
+                            );
+                        }
+                    } finally {
+                        onFinish();
+                    }
+                }
+
+                private void onFinish() {
+                    if (nextBatch.isEmpty()) {
+                        onFinish.close();
+                    } else {
+                        executeShardBulkInferenceAsync(inferenceId, inferenceProvider, nextBatch, onFinish);
+                    }
+                }
+            };
+            inferenceProvider.service()
+                .chunkedInfer(
+                    inferenceProvider.model(),
+                    null,
+                    inputs,
+                    Map.of(),
+                    InputType.INGEST,
+                    new ChunkingOptions(null, null),
```

Comment:
Seems like it would be better for `ChunkingOptions` to just be `null` no? It seems weird to pass around an object that is just `null` like this... Maybe its a inconsistency in the chunked infer api.

Reply:
That means default options - but agreed, it would be better to have a default constructor with no params here.